src/stores/BotCourtStore.js

```diff
@@ -0,0 +1,92 @@
+import { observable, action, runInAction, reaction } from 'mobx';
+import _ from 'lodash';
+import { Token, OracleStatus, AppLocation } from '../constants';
+import { queryAllOracles } from '../network/graphQuery';
+import Oracle from './models/Oracle';
+
+const INIT = {
+  loaded: false, // loading state?
+  loadingMore: false, // for laoding icon?
+  list: [], // data list
+  hasMore: true, // has more data to fetch?
+  skip: 0, // skip
+  limit: 16, // loading batch amount
+};
+
+export default class BotCourtStore {
+  @observable loaded = INIT.loaded
+  @observable loadingMore = INIT.loadingMore
+  @observable list = INIT.list
+  @observable hasMore = INIT.hasMore
+  @observable skip = INIT.skip
+  limit = INIT.limit
+
+  constructor(app) {
+    this.app = app;
+    reaction(
+      () => this.app.sortBy,
+      () => {
+        if (this.app.ui.location === AppLocation.botCourt) {
+          this.init(this.skip);
+        }
+      }
+    );
+    reaction(
+      () => this.list,
+      () => {
+        if (this.loaded && this.list.length < this.skip) this.hasMore = false;
+      }
+    );
+  }
+
+  @action
+  init = async (limit = this.limit) => {
+    this.reset(); // reset to initial state
+    this.app.ui.location = AppLocation.botCourt;
+    this.list = await this.fetch(limit);
+    runInAction(() => {
+      this.loading = false;
```

Comment:
supposed to be `this.loaded = true`?

Reply:
updated, here is where we change loaded, and hm sadly eslint didn't catch this kind, will be careful next time.